src/core/symbology/qgsfillsymbollayer.cpp

```diff
@@ -404,7 +404,19 @@ void QgsSimpleFillSymbolLayer::toSld( QDomDocument &doc, QDomElement &element, c
     // <Fill>
     QDomElement fillElem = doc.createElement( QStringLiteral( "se:Fill" ) );
     symbolizerElem.appendChild( fillElem );
-    QgsSymbolLayerUtils::fillToSld( doc, fillElem, mBrushStyle, mColor );
+    // Apply alpha from symbol
+    bool ok;
+    const double alpha { props.value( QStringLiteral( "alpha" ), 0.0 ).toDouble( &ok ) };
+    if ( ok )
+    {
+      QColor color { mColor };
+      color.setAlphaF( color.alphaF() * alpha );
+      QgsSymbolLayerUtils::fillToSld( doc, fillElem, mBrushStyle, color );
+    }
+    else
+    {
+      QgsSymbolLayerUtils::fillToSld( doc, fillElem, mBrushStyle, mColor );
```

Comment:
Just a minor comment for readability, could be remove the else part like it's done in the other modified part

Reply:
done